src/mig/modules/migoval/paramscreator.go

```diff
@@ -0,0 +1,105 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Contributor: Aaron Meihm ameihm@mozilla.com [:alm]
+package migoval
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+)
+
+func printHelp(isCmd bool) {
+	dash := ""
+	if isCmd {
+		dash = "-"
+	}
+	fmt.Printf(`Query parameters
+----------------
+%spkgmatch <string> - OS package search
+                    ex: pkgmatch linux-image
+		    query for installed OS packages containing substring
+
+%soval <path>       - OVAL processor
+		    ex: oval ./ovaldefs.xml
+		    process oval definitions on agent
+
+%snum <int>         - Concurrent OVAL checks (default: 1)
+                    ex: num 5
+		    Specify concurrent OVAL definitions to evaluate at once
+
+%sincludefalse      - Include false evaluations
+                    ex: includefalse
+		    Also includes definitions in results that evaluated to false
+`, dash, dash, dash, dash)
+}
+
+func (r Runner) ParamsParser(args []string) (interface{}, error) {
+	var (
+		fs           flag.FlagSet
+		ovalDefs     string
+		pkgMatch     flagParam
+		maxEval      int
+		includeFalse bool
+	)
+
+	if len(args) < 1 || args[0] == "" || args[0] == "help" {
+		printHelp(true)
+		return nil, nil
+	}
+
+	fs.Init("migoval", flag.ContinueOnError)
+	fs.Var(&pkgMatch, "pkgmatch", "see help")
```

Comment:
I'll take a look at this one when I add the other match options, I was going to look at matching on other modes then just substring (e.g., regex) so can look into this at that point to see how we want to structure it.